src/ErrorHandler.php

```diff
@@ -315,16 +341,21 @@ private function handleError(int $level, string $message, string $file, int $lin
      */
     private function handleFatalError(): void
     {
-        // If there is not enough memory that can be used to handle the error
-        // do nothing
-        if (null === self::$reservedMemory) {
-            return;
-        }
-
+        // Free the reserved memory that allows us to potentially handle OOM errors
         self::$reservedMemory = null;
+
         $error = error_get_last();
 
         if (!empty($error) && $error['type'] & (\E_ERROR | \E_PARSE | \E_CORE_ERROR | \E_CORE_WARNING | \E_COMPILE_ERROR | \E_COMPILE_WARNING)) {
+            // If we did not do so already and we are allowed to increase the memory limit, we do so when we detect an OOM error
+            if (false === self::$didIncreaseMemoryLimit && null !== $this->memoryLimitIncreaseValue && 1 === preg_match(self::OOM_MESSAGE_MATCHER, $error['message'], $matches)) {
```

Comment:
It's totally reasonable to limit it to just OOMs. However, I wonder if it is always acceptable to bump the memory, and then drop it back down after the handler runs.

Since this increase can be disabled/configured, it seems sane enough to me.

The main reason for suggesting this is to reduce the maintenance burden. You would not need regex matching.

I also understand if you'd rather not bump memory for all fatal errors.

Reply:
Yeah I thought about that, but only doing it when absolutely needed seemed like the best approach. If the regex turns out to be a mistake at a later point because the error message is changes we might change our stance for this. I think the fatal error always exits the process so it's probably safe to do it always, but I'm also not sure that is actually true :)